final-project-code/final-project/gsm/gsm.c

```diff
@@ -117,6 +119,16 @@ int gsmUtilDisassembleSms(char* meta, int offset)
 	}
 }
 
+void gsmutilGetSenderInfo(char* meta, char* phoneNumber)
+{
+	int k = 0;
+	int startIndex = 21;
+	int endIndex = startIndex + 11; 
+	for (int i = startIndex; i < endIndex ; i++)
```

Comment:
remove redundat space, line should be:

`for (int i = startIndex; i < endIndex; i++)`